src/main/scala/org/apache/spark/sql/delta/util/PartitionUtils.scala

```diff
@@ -577,14 +577,14 @@ private[delta] object PartitionUtils {
         // Then falls back to string
         .getOrElse {
         if (raw == DEFAULT_PARTITION_NAME) {
-          Literal.create(null, NullType)
+          Literal(null)
         } else {
           Literal.create(unescapePathName(raw), StringType)
         }
       }
     } else {
       if (raw == DEFAULT_PARTITION_NAME) {
-        Literal.create(null, NullType)
+        Literal(null)
```

Comment:
nit: Maybe it's better to keep the create() call in these two places? It makes it more clear that converting to catalyst types are needed here, although for null it's no-op